src/utils/MainApi.js

```diff
@@ -0,0 +1,101 @@
+class MainApi {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    } else {
+      return Promise.reject(`Main API Error Type: 
+${res.status} ${res.statusText}`);
+    }
+  }
+
+  ////////=============== AUTH ================= //////////
+
+  async signup({ email, password, name }) {
+    console.log(email, password, name);
+    const response = await fetch(`${this._baseUrl}/signup`, {
+      method: "POST",
+      headers: this._headers,
+      body: JSON.stringify({ email: email, password: password, name: name }),
+    });
+    return this._checkResponse(response);
+  }
+
+  signin({ email, password }) {
+    console.log("signin data:", email, password);
+    return fetch(`${this._baseUrl}/signin`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        Accept: "application/json",
+      },
+      body: JSON.stringify({ email: email, password: password }),
+    })
+      .then(this._checkResponse)
+      .then((data) => {
+        localStorage.setItem("jwt", data.token);
+        return data;
+      });
+  }
+
+  //Get user info from server:
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers,
+    });
```

Comment:
you need to check responsese with `.then(this._checkResponse)` everywhere inside `Api`